hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/fs/TestTrash.java

```diff
@@ -786,6 +793,69 @@ public void testTrashEmptier() throws Exception {
     emptierThread.join();
   }
 
+  /**
+   * Test trash emptier can whether delete non-checkpoint dir or not.
+   * @throws Exception
+   */
+  @Test
+  public void testTrashEmptierWithNonCheckpointDir() throws Exception {
```

Comment:
This new unit test seems not cover the case you try to fix, right? It could pass with/without this improvement.

Reply:
For this unit test, IMO, we should enable this config and then mkdir one path under Trash home directly, then wait to clean by Emptier. JFYI.